torch/nn/utils/clip_grad.py

```diff
@@ -46,7 +46,7 @@ def clip_grad_norm_(
         = _group_tensors_by_device_and_dtype([[g.detach() for g in grads]])  # type: ignore[assignment]
 
     if norm_type == inf:
-        norms = [g.detach().abs().max().to(first_device) for g in grads]
+        norms = [torch.linalg.norm(g.detach(), inf).to(first_device) for g in grads]
```

Comment:
I don't think `inf` has forech support quite yet.

Reply:
there's also this: 

https://github.com/pytorch/pytorch/blob/c9f4f01981fd73fcc7c27676cc50230cd1b5bc22/aten/src/ATen/native/ForeachOpsKernels.cpp#L321

Reply:
So I'm not really sure how everything is binded but the fast CUDA kernel doesn't seem to support values other than 1 or 2 for `norm_type`. 
https://github.com/pytorch/pytorch/blob/6e3e3dd477e0fb9768ee890ff6c6e5dd54aca679/aten/src/ATen/native/cuda/ForeachReduceOp.cu#L108-L114

If the frontend for `torch._foreach_norm` supports `inf` (typically defaults to slow version). I think we could remove the if/else conditions

Reply:
cc @janeyx99

Reply:
Okay then we need to remove the inf path right?